code/src/components/MessageList.js

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from 'react'
+
+import moment from 'moment'
+
+export const MessageList = () => {
+  const MESSAGES_URL = 'https://wk11livesession.herokuapp.com/messages'
+  const [messages, setMessages] = useState([]);
+
+  useEffect(() => {
+    fetchMessages()
+  }, [])
+
+    const fetchMessages = () => {
+    fetch(MESSAGES_URL)
+      .then((res) => {
+        return res.json();
+      })
+      .then((data) => {
+        data.reverse();
+
+        const filtredMessages = data.filter((message) => message.text)
+
+        setMessages(filtredMessages)
+      })
+}
+
+  const likeHearts = (messageId) => {
+    fetch(`https://happy-thoughts-technigo.herokuapp.com/thoughts/${messageId}/like`, {
```

Comment:
There is a POST 404 Bad Request error coming up when clicking on Heart button 💔
You have followed the right steps to implement this, the problem is that since you fetched messages from the wrong API, the data you got is not the correct one for this project. So you are not getting the correct message IDs.
In the console you'll see that this is the URL we are trying to send the POST request to in order to like a thought. We get error 404 because the URL is wrong.
It looks like this:
**https://happy-thoughts-technigo.herokuapp.com/thoughts/[object%20Object]/like**
Notice the place where there should be a messageID, we get this instead: **[object%20Object]** - so the POST does not go through.
Maybe try doing the fetch to the correct URL and it will probaby work as it should 💥